components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/test/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/RoleDAOTest.java

```diff
@@ -822,6 +881,39 @@ private RoleBasicInfo addRole(String roleName, String audience, String audienceI
         }
     }
 
+    private RoleBasicInfo addRole(String roleName, String audience, String audienceId, String tenantDomain,
+                                  List<Permission> permissions, List<String> userIDsList, List<String> userNamesList,
+                                  List<String> groupIDsList, Map<String, String> groupNamesMap, RoleDAOImpl roleDAO)
+            throws Exception {
+
+        OrganizationManager organizationManager = mock(OrganizationManager.class);
+        RoleManagementServiceComponentHolder.getInstance().setOrganizationManager(organizationManager);
+        lenient().when(organizationManager.getOrganizationNameById(anyString())).thenReturn("test-org");
+        lenient().when(organizationManager.resolveOrganizationId(anyString())).thenReturn(tenantDomain);
+        organizationManagementUtil.when(() -> OrganizationManagementUtil.isOrganization(anyString())).thenReturn(false);
```

Comment:
based on this mock, it seems this method is intended to create roles in root orgs.
Make that information available in the method name as well to avoid confusions and future errors

Reply:
Modified